examples/react/module-api-restful/src/components/user-list.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import { UserList } from "./user-list";
+
+const queryClient = new QueryClient({
+  defaultOptions: { queries: { retry: false } },
+});
+
+const wrapper = (props: { children: React.ReactNode }) => (
+  <QueryClientProvider client={queryClient}>
+    {props.children}
+  </QueryClientProvider>
+);
+
+describe("UserList", () => {
+  test("Should render user list when page loaded", async () => {
```

Comment:
[IMO] It would be nice to have tests for "error" and "no data" cases too, but it seems very low priority right now. We could create an issue and implement it another time!